### PR TITLE
[Feature][zeta][monitoring] Support exposing monitoring metrics by prometheus exporter protocol (#5070)

### PR DESCRIPTION
**Pull request by kim-up**
PR for  #5070 
The final effect ： https://github.com/kim-up/incubator-seatunnel/blob/telemetry/docs/en/seatunnel-engine/telemetry.md

## Overall Design
image

### Node Hazelcast Metrics
Getting metrics from `ManagedExecutorServiceMBean`

- systemExecutorMBean
- asyncExecutorMBean
- scheduledExecutorMBean
- clientExecutorMBean
- clientQueryExecutorMBean
- clientBlockingExecutorMBean
- queryExecutorMBean
- ioExecutorMBean
- offloadableExecutorMBean
- partitionServiceMBean

Including,  executor_executedCount、executor_isShutdown、executor_maxPoolSize 、executor_poolSize、executor_queueRemainingCapacity、executor_isShutdown、executor_queueSize

### Zeta Thread Pool Metrics
Getting metrics from `CoordinatorService#(ThreadPoolExecutor) executorService`
Including, pool_activeCount、pool_corePoolSize 、pool_maximumPoolSize、pool_poolSize、pool_completedTask_total、pool_task_total
### Zeta Job Metrics
Getting metrics from `CoordinatorService#(IMap<Object, Object>)runningJobInfoIMap`
Including, job count  in various states
### Node Jvm Metrics
Getting metrics from `io.prometheus.client.hotspot.prometheus.DefaultExports`
Including, Memory、BufferPools、Garbage、Thread、ClassLoading、VersionInf
### Cluster metrics
Getting metrics from `ClusterService`
 Including, cluster_info、cluster_time

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by liugddx** [comment]
Good job.

File: seatunnel-examples/seatunnel-engine-examples/src/main/java/org/apache/seatunnel/example/engine/SeaTunnelEngineExample.java [hunk 1]
```diff
@@ -23,23 +23,25 @@
...
 import java.nio.file.Paths;
 
 public class SeaTunnelEngineExample {
 
     public static void main(String[] args)
-            throws FileNotFoundException, URISyntaxException, CommandException {
-        String configurePath = args.length > 0 ? args[0] : "/examples/fake_to_console.conf";
+            throws IOException, URISyntaxException, CommandException {
+        String configurePath =
+                args.length > 0 ? args[0] : "/examples/stream_fakesource_to_file.conf";
```
Comment (liugddx) [inline seatunnel-examples/seatunnel-engine-examples/src/main/java/org/apache/seatunnel/example/engine/SeaTunnelEngineExample.java]: revert.

File: seatunnel-examples/seatunnel-engine-examples/src/main/java/org/apache/seatunnel/example/engine/SeaTunnelEngineExample.java [hunk 2]
```diff
@@ -23,23 +23,25 @@
...
         String configFile = getTestConfigFile(configurePath);
         ClientCommandArgs clientCommandArgs = new ClientCommandArgs();
         clientCommandArgs.setConfigFile(configFile);
         clientCommandArgs.setCheckConfig(false);
         clientCommandArgs.setJobName(Paths.get(configFile).getFileName().toString());
         // Change Execution Mode to CLUSTER to use client mode, before do this, you should start
         // SeaTunnelEngineServerExample
-        clientCommandArgs.setMasterType(MasterType.LOCAL);
+        clientCommandArgs.setMasterType(MasterType.CLUSTER);
```
Comment (liugddx) [inline seatunnel-examples/seatunnel-engine-examples/src/main/java/org/apache/seatunnel/example/engine/SeaTunnelEngineExample.java]: Same as above.

**Review by liugddx — COMMENTED** [review]
Revert example module.

**Comment by TyrantLucifer** [comment]
document has some dead links, please check.

image

**Comment by liugddx** [comment]
@ic4y @hailin0 @Hisoka-X @EricJoy2048 

This feature is not small, may need to discuss further development.

**Comment by liugddx** [comment]
> @ic4y @hailin0 @Hisoka-X @EricJoy2048
> 
> This feature is not small, may need to discuss further development.

@TyrantLucifer

**Comment by Hisoka-X** [comment]
> @ic4y @hailin0 @Hisoka-X @EricJoy2048
> 
> This feature is not small, may need to discuss further development.

Maybe we can mark it unstable. So we can change it after release. This just is an option I offered, not final decided.

File: seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/resources/hazelcast.yaml [hunk 3]
```diff
@@ -15,6 +15,7 @@
 # limitations under the License.
 #
 
+
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/resources/hazelcast.yaml]: Please revert this.

**Comment by hailin0** [comment]
> > @ic4y @hailin0 @Hisoka-X @EricJoy2048
> > This feature is not small, may need to discuss further development.
> 
> Maybe we can mark it unstable. So we can change it after release. This just is an option I offered, not final decided.

+1 
Good for quick experiments to improve it

**Comment by kim-up** [comment]
@liugddx @TyrantLucifer @Hisoka-X @hailin0 @EricJoy2048 PTAL

**Comment by ic4y** [comment]
Can we add the [slowoperation](https://docs.hazelcast.com/imdg/4.2/performance/slowoperationdetector) metric? It's quite valuable for troubleshooting. You can refer to the Hazelcast Management Center for guidance.

**Comment by kim-up** [comment]
> Can we add the [slowoperation](https://docs.hazelcast.com/imdg/4.2/performance/slowoperationdetector) metric? It's quite valuable for troubleshooting. You can refer to the Hazelcast Management Center for guidance.

Good idea! I will research it later. After the current PR is merged, create a new PR to improve it, because the current PR is relatively large.

**Comment by HoganEdwardChu** [comment]
We need this feature when does it merged?

**Comment by HoganChu98** [comment]
@TyrantLucifer @kim-up  
we need this feature does it planned to be merged? 
Thanks

**Comment by hailin0** [comment]
@kim-up 
good pr, please fix the conflict

**Comment by hailin0** [comment]
Please merge the dev branch and wait for ci to run successfully
@kim-up

**Comment by appleshcn** [comment]
@TyrantLucifer @kim-up @hailin0 
we need this feature too, does it planned to be merged?
Thanks

**Comment by emmanuelsosa** [comment]
@kim-up we’re excited to see this merged. Thank you for your contribution.

**Comment by Hisoka-X** [comment]
We should continue push this pr. @kim-up Could you fix conflict？ cc @EricJoy2048 @hailin0

**Comment by corgy-w** [comment]
@kim-up I intend to continue this feature of yours, as it has been too long since the last update. I will be creating a new branch and porting your code to implement the necessary changes. Please be aware of this and looking forward to cooperating with you,TKS! :>

**Pull request closed without merging**